src/mlpack/tests/main_tests/fastmks_test.cpp

```diff
@@ -0,0 +1,658 @@
+  indices = std::move(CLI::GetParam<arma::Mat<size_t>>("indices"));
+  kernel = std::move(CLI::GetParam<arma::mat>("kernels"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["reference"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["k"].wasPassed = false;
+
+  // Random input, some k <= number of reference points.
+  SetInputParam("reference", referenceData);
+  SetInputParam("k", (int) 10);
+  SetInputParam("naive", true);
+
+  mlpackMain();
+
+  CheckMatrices(indices,
+      CLI::GetParam<arma::Mat<size_t>>("indices"));
+  CheckMatrices(kernel,
+      CLI::GetParam<arma::mat>("kernels"));
+}
+
+/*
+ * Ensure that single-tree search returns the same result
+ * as dual-tree search
+ */
+BOOST_AUTO_TEST_CASE(FastMKSTreeTest)
+{
+  // 100 points in 3 dimensions.
+  arma::mat referenceData(3, 100, arma::fill::randu);
+
+  // Random input, some k <= number of reference points.
+  SetInputParam("reference", referenceData);
+  SetInputParam("k", (int) 10);
+
+  mlpackMain();
+
+  arma::Mat<size_t> indices;
+  arma::mat kernel;
+  indices = std::move(CLI::GetParam<arma::Mat<size_t>>("indices"));
+  kernel = std::move(CLI::GetParam<arma::mat>("kernels"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["reference"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["k"].wasPassed = false;
+
+  SetInputParam("reference", std::move(referenceData));
+  SetInputParam("k", (int) 10);
+  SetInputParam("single", true);
+
+  mlpackMain();
+
+  CheckMatrices(indices,
+      CLI::GetParam<arma::Mat<size_t>>("indices"));
+  CheckMatrices(kernel,
+      CLI::GetParam<arma::mat>("kernels"));
+}
+
+/*
+ * Ensure that we get almost same results in cover tree
+ * search mode when different basis is specified.
+ */
+BOOST_AUTO_TEST_CASE(FastMKSBasisTest)
+{
+  // 100 points in 3 dimensions.
+  arma::mat referenceData(3, 100, arma::fill::randu);
+
+  // Random input, some k <= number of reference points.
+  SetInputParam("reference", referenceData);
+  SetInputParam("k", (int) 10);
+  SetInputParam("base", 3.0);
+
+  mlpackMain();
+
+  arma::Mat<size_t> indices;
+  arma::mat kernel;
+  indices = std::move(CLI::GetParam<arma::Mat<size_t>>("indices"));
+  kernel = std::move(CLI::GetParam<arma::mat>("kernels"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["reference"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["k"].wasPassed = false;
+
+  SetInputParam("reference", std::move(referenceData));
+  SetInputParam("k", (int) 10);
+  SetInputParam("base", 4.0);
+
+  mlpackMain();
+
+  arma::Mat<size_t> newindices;
+  arma::mat newkernel;
+  newindices = std::move(CLI::GetParam<arma::Mat<size_t>>("indices"));
+  newkernel = std::move(CLI::GetParam<arma::mat>("kernels"));
+
+  CheckMatrices(indices, newindices);
+  CheckMatrices(kernel, newkernel);
+}
+
+/**
+  * Check that we can't specify base less than 2.
+  */
+BOOST_AUTO_TEST_CASE(FastMKSBaseTest)
```

Comment:
I don't think this should be true.  Please let me know why you added this test (maybe I overlooked something?).  I think that any base > 1 should be fine, but correct me if I'm wrong.

Reply:
Now this looks better, thanks for the fix. :+1: